From 2d75e2a226a6572041e2080e35c90bacad3656ae Mon Sep 17 00:00:00 2001
From: nferre <nferre@50fbe906-d41e-0410-8a96-31537896a350>
Date: Tue, 1 Jun 2010 13:57:27 +0000
Subject: [PATCH] at91: introduce at91sam9m10g45ek board

Slight differences oblige us to create a new board and mach-id for this
at91sam9m10g45 Evaluation Kit. Introduce the basic board file changes.



git-svn-id: svn://rfolxts01.rfo.atmel.com/at91_sandbox/linux-2.6.x/branches/linux-2.6.30-at91@15228 50fbe906-d41e-0410-8a96-31537896a350
---
 arch/arm/mach-at91/Kconfig              |   14 ++++++++++++--
 arch/arm/mach-at91/Makefile             |    1 +
 arch/arm/mach-at91/board-sam9m10g45ek.c |   12 ++++++++++--
 3 files changed, 23 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-at91/Kconfig b/arch/arm/mach-at91/Kconfig
index 04c6c4c..17e09cb 100644
--- a/arch/arm/mach-at91/Kconfig
+++ b/arch/arm/mach-at91/Kconfig
@@ -421,14 +421,24 @@ if ARCH_AT91SAM9M10
 
 comment "AT91SAM9M10 Board Type"
 
+choice 
+	prompt "Atmel Board"
+
 config MACH_AT91SAM9M10EKES
         bool "Atmel AT91SAM9M10-EKES Evaluation Kit"
-        depends on ARCH_AT91SAM9M10
         help
           Select this if you are using Atmel's AT91SAM9M10-EKES Evaluation Kit.
           "ES" at the end of the name means that this board is an
           Engineering Sample.
 
+config MACH_AT91SAM9M10G45EK
+        bool "Atmel AT91SAM9M10G45-EK Evaluation Kit"
+        help
+          Select this if you are using Atmel's AT91SAM9M10G45-EK Evaluation Kit.
+	  <http://atmel.com/dyn/products/tools_card_v2.asp?tool_id=4735>
+
+endchoice
+
 endif
 
 # ----------------------------------------------------------
@@ -489,7 +499,7 @@ config MTD_AT91_DATAFLASH_CARD
 
 config MTD_NAND_ATMEL_BUSWIDTH_16
 	bool "Enable 16-bit data bus interface to NAND flash"
-	depends on (MACH_AT91SAM9260EK || MACH_AT91SAM9261EK || MACH_AT91SAM9G10EK || MACH_AT91SAM9263EK || MACH_AT91SAM9G20EK || MACH_AT91SAM9G20EK_2MMC || MACH_AT91SAM9G45EKES || MACH_AT91CAP9ADK || MACH_AT572D940HFEB)
+	depends on (MACH_AT91SAM9260EK || MACH_AT91SAM9261EK || MACH_AT91SAM9G10EK || MACH_AT91SAM9263EK || MACH_AT91SAM9G20EK || MACH_AT91SAM9G20EK_2MMC || MACH_AT91SAM9G45EKES || MACH_AT91SAM9M10EKES || MACH_AT91SAM9M10G45EK || MACH_AT91CAP9ADK || MACH_AT572D940HFEB)
 	help
 	  On AT91SAM926x boards both types of NAND flash can be present
 	  (8 and 16 bit data bus width).
diff --git a/arch/arm/mach-at91/Makefile b/arch/arm/mach-at91/Makefile
index 5bb0f04..e42d82c 100644
--- a/arch/arm/mach-at91/Makefile
+++ b/arch/arm/mach-at91/Makefile
@@ -74,6 +74,7 @@ obj-$(CONFIG_MACH_SBC35_A9G20)	 += board-sbc35-a9g20.o
 # AT91SAM9G45 board-specific support
 obj-$(CONFIG_MACH_AT91SAM9G45EKES) += board-sam9m10g45ek.o
 obj-$(CONFIG_MACH_AT91SAM9M10EKES) += board-sam9m10g45ek.o
+obj-$(CONFIG_MACH_AT91SAM9M10G45EK) += board-sam9m10g45ek.o
 
 # AT91CAP9 board-specific support
 obj-$(CONFIG_MACH_AT91CAP9ADK)	+= board-cap9adk.o
diff --git a/arch/arm/mach-at91/board-sam9m10g45ek.c b/arch/arm/mach-at91/board-sam9m10g45ek.c
index 61a8076..0277e6b 100644
--- a/arch/arm/mach-at91/board-sam9m10g45ek.c
+++ b/arch/arm/mach-at91/board-sam9m10g45ek.c
@@ -2,6 +2,7 @@
  *  Board-specific setup code for the AT91SAM9M10G45 Evaluation Kit family
  *
  *  Covers: * AT91SAM9G45-EKES  board
+ *          * AT91SAM9M10-EKES  board
  *          * AT91SAM9M10G45-EK board
  *
  *  Copyright (C) 2009 Atmel Corporation.
@@ -253,9 +254,14 @@ static struct atmel_lcdfb_info __initdata ek_lcdc_data;
  * Touchscreen
  */
 static struct at91_tsadcc_data ek_tsadcc_data = {
+#if defined(CONFIG_MACH_AT91SAM9M10G45EK)
+	.adc_clock		= 800000,
+	.ts_sample_hold_time	= 0x04,
+#else
 	.adc_clock		= 300000,
-	.pendet_debounce	= 0x0d,
 	.ts_sample_hold_time	= 0x0a,
+#endif
+	.pendet_debounce	= 0x0d,
 };
 
 
@@ -427,8 +433,10 @@ static void __init ek_board_init(void)
 
 #if defined(CONFIG_MACH_AT91SAM9G45EKES)
 MACHINE_START(AT91SAM9G45EKES, "Atmel AT91SAM9G45-EKES")
-#else
+#elif defined(CONFIG_MACH_AT91SAM9M10EKES)
 MACHINE_START(AT91SAM9M10EKES, "Atmel AT91SAM9M10-EKES")
+#else
+MACHINE_START(AT91SAM9M10G45EK, "Atmel AT91SAM9M10G45-EK")
 #endif
 	/* Maintainer: Atmel */
 	.phys_io	= AT91_BASE_SYS,
-- 
1.5.6.5

